### PR TITLE
[WIP]AdminUI

### PR DESCRIPTION
**Pull request by codingfabi**
!!!Please note that this is highly WIP, I opened a pull requests because I could need some help and another pair of eyes to look at it!!
see issue: #254 

### Visible/Frontend Changes
- [x] List of all curent Users with CRUD functionality, only accessable for admins

### Architectural/Backend Changes
- [x] enabling storing of users
- [x] enabling deleting of users
- [x] enabling user loading from database
- [ ] enabling updating users including password

### CODE REVIEW

**Comment by codingfabi** [comment]
Let me explain the wokraround (commit 7e8d06c) because it may seem like the dummest thing in existence but it actually is necessary:

`generateNewRecordFactory(){
    return () => {
      const newUser= new User('');
      return newUser;
    };
  }`
when creating new entitys with the entity-subrecord component, we need to initilize a new entity before we edit the rest of the information. in case of the user, the id is dependant on the username and therfore dependant on the admins input for the username. 

so i tried to catch the creation of a new user in the entity subrecord, but it is not possible to get to the password the admin has entered. therefore, i had to do this piece of whatever you want to call it just to get the whole thing running:
`if(record.getType()==='User'){
      let user: User;
      user=new User((record as User).name);
      user.setNewPassword('pass');
      user.admin=(record as User).admin;
      user.name=(record as User).name;
      record=user;
    }`

so to get it running, it was enough but it is basically unusable. 
See Issue #254 for further discussions

**Comment by sleidig** [comment]
As the user management with passwords is a somewhat special case, it might be easiest to implement it's own template without EntitySubrecordComponent. In fact even ChildrenList and SchoolList don't use a common component yet - although that's on the roadmap (#199).

Otherwise, we could think about extending `ColumnDescription` with another attribute, a "saveFunction" that defines a special way how to write the field? Or could we make the `setPassword()` function of User into a simple setter (that still converts the plaintext password into the encrypted password object)?

File: src/app/ui-helper/entity-subrecord/entity-subrecord.component.ts [hunk 1]
```diff
@@ -49,8 +50,22 @@ export class EntitySubrecordComponent implements OnInit, OnChanges {
...
+  /**
+   * 
+   * @param record uses the input from the form field to create a record
+   * curently has ugly workaround for entity type user because the id is dependent from the users input
+   * this workaround is highly WIP because its buggy aswell for creating a user twice somehow. 
+   */
   save(record: Entity) {
+    if(record.getType()==='User'){
+      let user: User;
+      user=new User((record as User).name);
```
Comment (sleidig) [inline src/app/ui-helper/entity-subrecord/entity-subrecord.component.ts]: Don't you lose the "_rev" field of an existing user object when creating a new user like this? That would make PouchDB reject changing it.

**Comment by TheSlimvReal** [comment]
As far as I got the state of the PR what needs to be done is to correlctly implement the `signupUser` function for the Pouch and probably also the moch database. Added to that I think there is a more elegant solution to the `save` method in the entity subrecord component. As @sleidig mentiond, it might makes sense to use a custom component here.

**Comment by TheSlimvReal** [comment]
Closing in favor of #389

**Pull request closed without merging**